src/aviary/utils.py

```diff
@@ -510,3 +512,50 @@ def format_exc(exc: BaseException) -> str:
             f" {', '.join(repr(e) for e in exc.exceptions)}"
         )
     return repr(exc)
+
+
+class ReaderWriterLock:
```

Comment:
May be nice to just subclass a `Lock` class here to match the interface. And feel free to ignore this one